### PR TITLE
fix: (farms) Error when put amount with decimals more than 18

### PR DESCRIPTION
**Pull request by memoyil**
To review:

https://deploy-preview-1446--pancakeswap-dev.netlify.app/

To Reproduce the issue:

1. Go to farms
2. Click withdraw on any farm that you staked
3. Put amount with more than 20 decimals
4. Check error on console

### CODE REVIEW

File: src/components/ModalInput/ModalInput.tsx [hunk 1]
```diff
@@ -64,6 +65,7 @@ const ModalInput: React.FC<ModalInputProps> = ({
   value,
   addLiquidityUrl,
   inputTitle,
+  decimals = 20,
```
Comment (hachiojidev) [inline src/components/ModalInput/ModalInput.tsx:68]: Should be 18

File: src/components/ModalInput/ModalInput.tsx [hunk 2]
```diff
@@ -88,7 +90,7 @@ const ModalInput: React.FC<ModalInputProps> = ({
         </Flex>
         <Flex alignItems="flex-end" justifyContent="space-around">
           <StyledInput
-            pattern="^[0-9]*[.,]?[0-9]*$"
+            pattern={`^[0-9]*[.,]?[0-9]{0,${decimals}}$`}
```
Comment (hachiojidev) [inline src/components/ModalInput/ModalInput.tsx:93]: This pattern (and decimal prop) should be updated on the UI Kit and removed from here.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/components/ModalInput/ModalInput.tsx (hunk 1 above)
Comment (memoyil) [reply src/components/ModalInput/ModalInput.tsx]: done

File: src/components/ModalInput/ModalInput.tsx (hunk 2 above)
Comment (memoyil) [reply src/components/ModalInput/ModalInput.tsx:93]: I opened a pr for BalanceInput in UI Kit

https://github.com/pancakeswap/pancake-toolkit/pull/142

Comment (memoyil) [reply src/components/ModalInput/ModalInput.tsx:93]: Input component is used in many places for different reasons in frontend (such as search input for text, lottery search for integer and transfer nft for wallet address), if we move this pattern to uikit we need to refactor all of the usages and write rules for each of them.

Comment (hachiojidev) [reply src/components/ModalInput/ModalInput.tsx:93]: Yes, good catch.

Comment (hachiojidev) [reply src/components/ModalInput/ModalInput.tsx:93]: Wait, you added it to the BalanceInput which is fine.

Comment (memoyil) [reply src/components/ModalInput/ModalInput.tsx:93]: I think it is ready to merge wdyt?

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**